tasks/lib/code-coverage-enforcer-lib.js

```diff
@@ -121,13 +121,14 @@ module.exports = (function () {
      * @param {array}  files   a pointer to an array that stores a list of files
      * @param {array}  includes a list of patterns for files to match
      * @param {array}  excludes a list of patterns for files to ignore
+     * @param {string} extension    an optional regular expression for matching file extensions
      *
      * collect(options.src, fileList, options.includes, options.excludes, null);
      */
-    exports.collect = function (fp, files, includes, excludes, replaceDirectory) {
+    exports.collect = function (fp, files, includes, excludes, replaceDirectory, extension) {
         if (shjs.test("-d", fp)) {
             shjs.find(fp).filter(function (file) {
-                if (file.match(/\.js$/)) {
+                if (file.match(extension || /\.js$/)) {
```

Comment:
I wonder if saying /\.jsx$|\.js$/ is preferable to adding a boolean here? 

I.e. have a single regex.

http://stackoverflow.com/questions/4029109/javascript-regex-how-to-put-a-variable-inside-a-regular-expression